libfreerdp/codec/progressive.c

```diff
@@ -1755,11 +1780,11 @@ int progressive_decompress(PROGRESSIVE_CONTEXT* progressive, BYTE* pSrcData, UIN
 				if ((blockLen - boffset) < region->tileDataSize)
 					return -1021;
 
-				if (region->numTiles > progressive->cTiles)
+				if (progressive->cTiles < surface->gridSize)
 				{
 					progressive->tiles = (RFX_PROGRESSIVE_TILE**) realloc(progressive->tiles,
-							region->numTiles * sizeof(RFX_PROGRESSIVE_TILE*));
-					progressive->cTiles = region->numTiles;
+							surface->gridSize * sizeof(RFX_PROGRESSIVE_TILE*));
+					progressive->cTiles = surface->gridSize;
```

Comment:
I suggest to simply allocate progressive->tiles in progressive_surface_context_new() instead and kick progressive->cTiles